src/commands/test/test_subject_queue.py

```diff
@@ -0,0 +1,32 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING, List, Tuple
+
+from src.commands.test.test_cases import TestCaseResult
+from src.utils.starknet_compilation import StarknetCompiler
+
+if TYPE_CHECKING:
+    import queue
+
+
+@dataclass
+class TestSubject:
+    """
+    A dataclass consisting of identification of a single test bundle, and target functions
+    """
+
+    test_path: Path
+    test_functions: List[StarknetCompiler.AbiElement]
+
+
+class TestSubjectQueue:
+    def __init__(
+        self, shared_queue: "queue.Queue[Tuple[TestSubject, TestCaseResult]]"
+    ) -> None:
+        self._shared_queue = shared_queue
+
+    def dequeue(self) -> Tuple[TestSubject, TestCaseResult]:
```

Comment:
put, get are idiomatically used in Python queues

Reply:
IMO `get` is not obvious whether it modifies the queue or not. I'll rename it to match Python though.